### DISCUSSION TITLE
Chinese chars are not displayed by cog

### DISCUSSION THREAD

**Discussion by WoeMat** [Igalia/cog#458] [Q&A] [answered]
Hi.
We are using cog to display our UI which is an angular web app. Right now we want to add a chinese translation to this app. Cog is therefore running on a control panel which is using a touchscreen and a Raspberry CM3.
If I load the website with my desktop browser I can see all the chinese chars but when it is displayed by cog on the control panel I can only see rectangulas instead of chinese chars.
Is there any way to enable chinese chars? I allready tried to pass the option --default-charset="utf8" but it had no efect.

Thanks

**Comment by philn** [comment]
Here on my desktop with Cog/GTK4, chinese glyphs from https://en.wikipedia.org/wiki/Chinese_language seem to render fine. Maybe you're missing some fonts on your rpi OS?

**Comment by tobias-bischoff** [comment, marked as answer]
@WoeMat For embedded use-cases, it's best to ship a Web font with your application.

**Reply by WoeMat** [reply]
Thanks for the help. I found the corresponding fonts :)
